### PR TITLE
Allow dynamodb cloud usage

### LINKED ISSUE #3442: DynamoDB KV Support - Integrate with Production DynamoDB
DoD: A running instance/cluster of DynamoDB. KV DynamoDB implementation can connect to that DB and run properly.

Part of this task is to figure out various aspects of production dynamoDB usage:
* read/write capacities
* DyamoDB cluster structure
* AWS parameters (TBD?)

### PR DESCRIPTION
**Pull request by itaiad200**
closes #3442

### CODE REVIEW

**Review by N-o-Z — APPROVED** [review]
Overall look good. One comment though - you need to modify the predicate handling as well and change it to byte slice (blob)

File: pkg/kv/dynamodb/store.go [hunk 1]
```diff
@@ -165,11 +173,33 @@ func setupKeyValueDatabase(ctx context.Context, svc *dynamodb.DynamoDB, params *
 		},
 	})
 	if err != nil {
-		if _, ok := err.(*dynamodb.ResourceInUseException); !ok {
-			return err
+		if _, ok := err.(*dynamodb.ResourceInUseException); ok {
```
Comment (N-o-Z) [inline pkg/kv/dynamodb/store.go:176]: if err != nil and err is ResourceInUseException (table already created - as in the case of open twice) we should not wait for table ACTIVE state

**Review by N-o-Z — CHANGES_REQUESTED** [review]

File: pkg/kv/dynamodb/store.go [hunk 2]
```diff
@@ -192,8 +222,9 @@ func (s *Store) Get(ctx context.Context, partitionKey, key []byte) (*kv.ValueWit
 	}
 
 	result, err := s.svc.GetItemWithContext(ctx, &dynamodb.GetItemInput{
-		TableName: aws.String(s.params.TableName),
-		Key:       s.bytesKeyToDynamoKey(partitionKey, key),
+		TableName:      aws.String(s.params.TableName),
+		Key:            s.bytesKeyToDynamoKey(partitionKey, key),
+		ConsistentRead: aws.Bool(true),
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/kv/dynamodb/store.go:227]: Yap - there's no other option other for lakeFS than being consistent. lakeFS provides ACID guarantees and not being consistent while reading from any DB breaks that promises. For example: You committed by advancing the branch to the next staging token. Some read after that still reads the previous staging token after the commit already happened. Now you write a file under a stale staging token and it's forever gone.

File: pkg/kv/dynamodb/store.go (hunk 1 above)
Comment (itaiad200) [reply pkg/kv/dynamodb/store.go:176]: Which is what I believe I did, no?

**Pull request merged**